### PR TITLE
[Update] Add components of authors list, author page, and sponsor

### PR DESCRIPTION
**Pull request by YuCJ**
- Add authors list with searching function (search by name)
- Add single author data page and related articles
- Add sponsor component
- Fetch authors data from Algolia (through our own API server with Redis cache)

### CODE REVIEW

File: api/constants.js [hunk 1]
```diff
@@ -1,3 +1,8 @@
 export default {
   timeout: 1000
 }
+
+export const APP_ID = 'CTAAU9LLQ2'
+export const SEARCH_API_KEY = '4e4d99db26090df488e481abf441914d'
```
Comment (garfieldduck) [inline api/constants.js]: take APP_ID and SEARCH_API_KEY  into `api/config.js`

File: src/components/authorPage/AuthorData.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+const AuthorData = (props) => {
+  const authorData = props.authorData
+  let gotAuthorMail = !(authorData.authorMail === undefined) ? true : false
+  return (
+  <div className={boxClasses}>
+    <div className={styles['author-img-container']}><img className={styles['author-img']} src={authorData.authorImg}/></div>
+    <div className={styles['author-data-container']}>
+      <div className={styles['author-name']}>{authorData.authorName}</div>
+      {!gotAuthorMail ? null : <div className={styles['author-mail']}>{authorData.authorMail}</div>}
```
Comment (garfieldduck) [inline src/components/authorPage/AuthorData.js]: use `!authorData.authorMail` instead (?)
```
{!authorData.authorMail ? null : <div className={styles['author-mail']}>{authorData.authorMail}</div>}
```

File: src/components/authors/AuthorSearchBox.scss [hunk 3]
```diff
@@ -0,0 +1,188 @@
+@import '../../themes/common-variables';
+
+$placeholder-color: #A4A4A4;
+$input-bg-color: #F9F9F9;
+$input-text-color: #CBCBCB;
+$searchbox-height: 30px;
+$searchbox-width: 250px;
+$font-size: 15px;
```
Comment (garfieldduck) [inline src/components/authors/AuthorSearchBox.scss]: extract $font-size into `common-variables.scss`

File: src/components/authors/AuthorSearchBox.scss [hunk 4]
```diff
@@ -0,0 +1,188 @@
+@import '../../themes/common-variables';
+
+$placeholder-color: #A4A4A4;
+$input-bg-color: #F9F9F9;
+$input-text-color: #CBCBCB;
```
Comment (garfieldduck) [inline src/components/authors/AuthorSearchBox.scss]: put color definitions into `common-variables.scss`

File: src/components/authors/AuthorSearchBox.scss [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+
+  &:hover {
+    box-shadow: inset 0 0 0 0.5px #b2b2b2;
+  }
+
+  &:focus,
+  &:active,{
+    outline: 0;
+    box-shadow: inset 0 0 0 0.5px $input-text-color;
+    background: #FFFFFF;
```
Comment (garfieldduck) [inline src/components/authors/AuthorSearchBox.scss]: $white

File: src/components/authors/AuthorSearchBox.scss [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+
+  &:focus {
+    outline: 0;
+  }
+
+  & img {
+    width: 15px;
+    height: 15px;
+    vertical-align: middle;
+    fill: #CBCBCB;
```
Comment (garfieldduck) [inline src/components/authors/AuthorSearchBox.scss]: put it into `common-variables.scss` (?)

File: src/components/authors/AuthorSearchBox.scss [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+    display: block;
+    margin: 4px;
+    width: 9px;
+    height: 9px;
+  }
+}
+
+.input-box:valid ~ .reset-button {
+  display: block;
+  -webkit-animation-name: reset-in;
```
Comment (garfieldduck) [inline src/components/authors/AuthorSearchBox.scss]: you may not need to add vendor prefix like `-webkit-`

File: src/themes/animation.scss [hunk 8]
```diff
@@ -59,6 +59,16 @@
   animation-name: fade-in-down;
 }
 
+// Fade-in-down-wait
+@-webkit-keyframes fade-in-down { 0% { opacity: 0; -webkit-transform: translateY(-20px); } 100% { opacity: 1; -webkit-transform: translateY(0); } }
```
Comment (garfieldduck) [inline src/themes/animation.scss]: you may not need to add vendor prefix like `-webkit-`

**Comment by YuCJ** [comment]
**[Update]**
- Move Algolia key and id from constants to config
- Make author data parser more concise
- Update css styles
- Add Link to the HeadingAuthor of each article

**Comment by YuCJ** [comment]
**[Update]**

- [Styles] Fix AuthorCollection styles on mobile
- [Styles] Fix AuthorData  text align of authorBio on mobile
- Update author default img

**Pull request merged**